Jetcaster/app/src/main/java/com/example/jetcaster/ui/player/PlayerScreen.kt

```diff
@@ -111,41 +122,113 @@ private fun PlayerScreen(
 @Composable
 fun PlayerContent(
     uiState: PlayerUiState,
+    windowInfo: WindowInfo,
     onBackPress: () -> Unit
 ) {
     PlayerDynamicTheme(uiState.podcastImageUrl) {
+        // As the Player UI content changes considerably when the device is in tabletop posture,
+        // we split the different UIs in different composables. For simpler UIs that don't change
+        // much, prefer one composable that makes decisions based on the mode instead.
+        if (windowInfo.isInTableTopPosture) {
+            PlayerContentTableTop(uiState, windowInfo, onBackPress)
+        } else {
+            PlayerContentRegular(uiState, onBackPress)
+        }
+    }
+}
+
+@Composable
+private fun PlayerContentRegular(
+    uiState: PlayerUiState,
+    onBackPress: () -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalGradientScrim(
+                color = MaterialTheme.colors.primary.copy(alpha = 0.50f),
+                startYPercentage = 1f,
+                endYPercentage = 0f
+            )
+            .systemBarsPadding()
+            .padding(horizontal = 8.dp)
+    ) {
+        TopAppBar(onBackPress = onBackPress)
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier.padding(horizontal = 8.dp)
+        ) {
+            Spacer(modifier = Modifier.weight(1f))
+            PlayerImage(
+                podcastImageUrl = uiState.podcastImageUrl,
+                modifier = Modifier.weight(10f)
+            )
+            Spacer(modifier = Modifier.height(32.dp))
+            PodcastDescription(uiState.title, uiState.podcastName)
+            Spacer(modifier = Modifier.height(32.dp))
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier.weight(10f)
+            ) {
+                PlayerSlider(uiState.duration)
+                PlayerButtons(Modifier.padding(vertical = 8.dp))
+            }
+            Spacer(modifier = Modifier.weight(1f))
+        }
+    }
+}
+
+@Composable
+private fun PlayerContentTableTop(
+    uiState: PlayerUiState,
+    windowInfo: WindowInfo,
+    onBackPress: () -> Unit
+) {
+    val hingeRect: Rect = windowInfo.hingePosition
+        ?: throw IllegalStateException("Bounds should never be null in tabletop mode")
+
+    val hingePosition = with(LocalDensity.current) { hingeRect.top.toDp() }
+    val hingeHeight = with(LocalDensity.current) { (hingeRect.bottom - hingeRect.top).toDp() }
+
+    Column(modifier = Modifier.fillMaxSize()) {
+        // Content for the top part of the screen
         Column(
             modifier = Modifier
-                .fillMaxSize()
+                .height(hingePosition)
```

Comment:
This works fine here, since the `PlayerContentTableTop`'s bounds match the window bounds exactly (due to `WindowCompat.setDecorFitsSystemWindows`.

I wonder what it would take to align the hinge properly if that wasn't the case, using some sort of bound adjustment logic.

Reply:
Yeah, that's a concern to have, that's why I have this logic inside `PlayerContentTableTop` and I don't provide something like a `TableTopLayout`. Btw, I provided it originally but exactly because of this, I removed it because I didn't want to make it that copy-pasteable